src/main/java/com/enderio/core/client/gui/GuiContainerBase.java

```diff
@@ -201,9 +203,92 @@ protected boolean func_146978_c(int p_146978_1_, int p_146978_2_, int p_146978_3
     return ghostSlots;
   }
 
+  /**
+   * Called when a ghost slot is clicked or mouse wheeled.
+   * 
+   * @param slot
+   *          The GhostSlot
+   * @param x
+   *          Mouse position x
+   * @param y
+   *          Mouse position y
+   * @param button
+   *          The button used (0=left, 1=right). The mouse wheel is mapped to
+   *          -1=down and -2=up.
+   */
   protected void ghostSlotClicked(GhostSlot slot, int x, int y, int button) {
```

Comment:
As most of it are the conditions with at most 3 lines in a block, I see no good (natural) way to split it up. Putting each button in its own method would work, but wouldn't really add readability---that there are 4 blocks without overlap can be seen at first glance.